From fb984aaa1e8933cbe08820b9adf51eab3f1825d0 Mon Sep 17 00:00:00 2001
From: Assam Boudjelthia <assam.boudjelthia@qt.io>
Date: Thu, 22 Oct 2020 18:29:47 +0300
Subject: [PATCH 125/171] Android: print tailored warning if qml dependency
 path is a dir

androiddeployqt might be misleading when the path is adir
and the warning says that the file does not exist.

Change-Id: I1129f49af58a0637a240fcfd425a61b2ed15c840
Reviewed-by: Ville Voutilainen <ville.voutilainen@qt.io>
(cherry picked from commit c730a29260eda298683803bfd5306926696eedb0)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/tools/androiddeployqt/main.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/tools/androiddeployqt/main.cpp b/src/tools/androiddeployqt/main.cpp
index 22facce703..b111cfba28 100644
--- a/src/tools/androiddeployqt/main.cpp
+++ b/src/tools/androiddeployqt/main.cpp
@@ -1798,6 +1798,12 @@ bool scanImports(Options *options, QSet<QString> *usedDependencies)
 
             QFileInfo info(path);
 
+            if (info.isDir()) {
+                if (options->verbose)
+                    fprintf(stdout, "    -- Skipping because path is a directory.\n");
+                continue;
+            }
+
             // The qmlimportscanner sometimes outputs paths that do not exist.
             if (!info.exists()) {
                 if (options->verbose)
-- 
2.31.1

